### DISCUSSION TITLE
LSP strange behaviour on completion

### DISCUSSION THREAD

**Discussion by zoedsoupe** [helix-editor/helix#8706] [Q&A] [answered]
I’m using helix `v23.10` and this only happens with `lexical-lsp` and `next-ls` for `elixir` language.

I know that i can pres `<ret>` to confirm completion and `tab` and `shift+tab` to cycle trough completions list. However this happens when I try top confirm a lexical-lsp completion:

 

https://github.com/helix-editor/helix/assets/44469426/702118ac-2d3f-41c5-822c-b84cee9399e2

**Comment by pascalkuthe** [comment]
what exactly is the issue here?  I don't see anything wrong with that video?

**Reply by zoedsoupe** [reply]
When I “hover” trough a completion option, it inputs on the buffer a “snippet” for the argumetns function, like the `String.trim/1` function as:

```elixir
String.trim(string)
```

However when i press `<ret>` the resulting completion is

```elixir
# cursor inside parens
String.trim()
```

For 1 arity functions that’s not a problem, however for 2 or more it results:

```elixir
# writes “,” to buffer but not argument names?
String.at(, ,)
```

the acceptance of the completion would also benefit the struct fields completion, like:

```elixir
# showed by “hover” on completion option:
%MYStruct{foo: foo}

# inserted on acceptance:
%MyStruct{foo: }
```

it’s seems kinda confusing to me 🤔

**Reply by pascalkuthe** [reply, marked as answer]
those are snippept placeholder. They are supposed to be removed automatically according to the lsp spec. In some languages they can be useful to keep but not in others. Once we support placeholders properly we can make them work like vscode where you can keep the content by hitting tab again (in vscode they disappear as soon as you start typing) but that is not implemented yet.
